### PR TITLE
[CAST-209] Display GeoIP Analysis in CAST Frontend

### PR DESCRIPTION
**Pull request by justincorsha**
Before you submit your PR, make sure you check each of the following:

1. [x] I have read the [Contributing Guidelines](./CONTRIBUTING.md) and signed the [Contributor License Agreement](./CLA.md).
2. [x] My code is clean and ready-to-PR, including being free of lint errors and dead code
3. [x] I have **tested** my change / code and am confident that it works
4. [x] I have filled in the template below completely and accurately
##### Changelog

- Updated ReusedAuthenticationCard to have a mui Table with collapsible rows that provide the location for each request for a given secret
- Updated the reused auth query to only return relevant data for the outer table
- Added URI, port and occurredAt to the LocationData schema to be provided as data to the inner tables
- Changed joins in geoip table to be left joins to allow for all requests to be populated in the inner table (including those will null location data such as local ips)

##### Testing

- Deployed CAST with sample data from `generate-pipeline-data.sh` and saw that it was populated in the Reused Authentication table
- Checked that requests are shown when geoip data is not imported

https://github.com/corshatech/cast/assets/109618849/fc6a52cc-2943-40f8-a440-5b88d5531c73

##### Unit Tests

N/A

### CODE REVIEW

File: ui/components/app/Analysis/ReusedAuthenticationCard.tsx [hunk 1]
```diff
@@ -9,40 +9,153 @@
...
-const columns: GridColDef[] = [
-  { field: 'Secret', headerName: 'Secret', width: 400 },
-  { field: 'Src IP', headerName: 'Src IP' },
-  { field: 'URI', headerName: 'URI', width: 400 },
-  { field: 'Dest IP', headerName: 'Dest IP' },
-  { field: 'Dest Port', headerName: 'Dest Port' },
-];
+// Severity of geoip distance in kilometers
+export const CRITICAL_SEVERITY_DISTANCE = 1000;
+export const HIGH_SEVERITY_DISTANCE = 100;
```
Comment (ArcticLight) [inline ui/components/app/Analysis/ReusedAuthenticationCard.tsx:34]: Can these constants please be moved to `@/lib/findings`?

**Review by ArcticLight — CHANGES_REQUESTED** [review]
Just the one nit, great job

**Review by ArcticLight — CHANGES_REQUESTED** [review]

File: ui/components/app/Analysis/ReusedAuthenticationCard.tsx [hunk 2]
```diff
@@ -9,40 +9,155 @@
...
+const Row = ({row}: {row: RequestData}) => {
+  const [open, setOpen] = useState(false);
+  return (
+    <>
+      <TableRow>
+        <TableCell>{row['Secret']}</TableCell>
+        <TableCell>{row['URI']}</TableCell>
+        <TableCell>{row['Count']}</TableCell>
+        <TableCell>
+          <IconButton aria-label="expand row" onClick={() => setOpen(!open)}>
```
Comment (JasonCorsha) [inline ui/components/app/Analysis/ReusedAuthenticationCard.tsx:114]:
```suggestion
          <IconButton aria-label="expand row" onClick={() => setOpen((prev) => !prev)}>
```

**Review by JasonCorsha — APPROVED** [review]
LGTM

File: ui/lib/analysis/reused-authentication.ts [hunk 3]
```diff
@@ -176,10 +177,17 @@ function rowsToFinding(detectedAt: string, reusedAuthRow: ReusedAuthRow, request
...
+      maxDist >= CRITICAL_SEVERITY_DISTANCE ?
+        'critical'
+      : 'high'
+    : 'medium';
+
   return {
     type: 'reused-auth',
     name: 'Broken Authentication: Reused Authorization',
-    severity: 'medium',
+    severity: severity,
```
Comment (ArcticLight) [inline ui/lib/analysis/reused-authentication.ts:190]: Nit: Prefer puns when possible for readability

```suggestion
    severity,
```

**Review by ArcticLight — CHANGES_REQUESTED** [review]

**Review by ArcticLight — APPROVED** [review]

**Pull request merged**